src/main/java/org/jvnet/hudson/update_center/Main.java

```diff
@@ -341,7 +355,18 @@ private String prettyPrintJson(JSONObject json) {
     }
 
     protected MavenRepository createRepository() throws Exception {
-        MavenRepository repo = DefaultMavenRepositoryBuilder.getInstance();
+
+        MavenRepositoryImpl base = DefaultMavenRepositoryBuilder.getInstance();
+        if (javaVersion != null) {
+            base.addPluginFilter(new JavaVersionPluginFilter(new VersionNumber(javaVersion)));
+        } else {
+            System.out.println("WARNING: Target Java version is not defined, version filters will not be applied");
+            //TODO: Default to the version actually supported by the target core if CAP is set?
```

Comment:
this prevents impact on the common and stable update centers.
I would like to enforce checks there, but it feels like a potentially breaking change since plugin POM does not really control correctness of the setting IIRC

Reply:
L160 above says we default to Java 8, it does not look like it is the case actually, is it?